### ISSUE TITLE
Technology planning 

### ISSUE THREAD

**Issue by venikman** [codepatch-school/codepatch.io#1]
Let's make a decision about what technologies we going to use to build our web-app. 
My thoughts is: 
## Server
* Serverless
* GraphQL
* Firebase

## Client
* Apollo for GraphQL
* React
* Redux
* Any CSS framework? (SASS...)

@sholladay your thoughts?

**Comment by sholladay** [comment]
I am on board with trying GraphQL and Firebase. I'm not so sure about serverless yet. It seems very bleeding edge and rough to me. Which serverless framework would you want to use?

Some serverless resources:

 - [About serverless systems](https://hackernoon.com/a-crash-course-on-serverless-with-node-js-632b37d58b44)
 - [webtask.io](https://webtask.io) or [AWS Lambda](https://aws.amazon.com/lambda/) for running functions as a service
 - [Comparison of Webtask vs Lambda](https://thenewstack.io/often-choose-webtask-lambda/)
 - Node.js ["serverless" framework](https://github.com/serverless/serverless) for building serverless apps

**Comment by venikman** [comment]
Agree. I don't see benefit (maybe my low knowledge). We could stick to Firebase and maybe GraphQL also I am interested in using TypeScript. What do you think? @sholladay 
We also can start use only semantic-ui because we don't care strict about our design we can use style out of the box.

**Comment by sholladay** [comment]
After my last comment, I discovered [StdLib](https://stdlib.com/), a serverless system that seems nice. They have some kind of [Stripe integration](https://hackernoon.com/build-a-serverless-stripe-store-in-5-minutes-with-node-js-and-stdlib-d90352c89227) that also looks interesting.

As for TypeScript, I like it, and we could use it, but I think we shouldn't teach it to beginner programmers. Strict typing is yet another thing to learn about for beginners and probably makes it a bit harder overall. I think plain JavaScript is good for learning, whereas TypeScript is good for building real apps, especially when the team has > 5 developers. So I suppose the answer is mostly based on what type of content we want to make. Personally, I am pretty interested in content for beginners. But ideally we would also have some advanced content so that people could stay with us as they gain experience. Maybe some of the advanced content should cover TypeScript.

Semantic UI I would definitely love to use! In fact, when we can't find a component we need in Semantic, maybe we should try to make a pull request there to add it. I'm sure we'll run into these and we can turn them into lessons that teach people how to make pull requests, how to code UI components, how to deal with PR feedback, etc.

**Comment by sholladay** [comment]
Let's use [Next.js](https://nextjs.org/) for rendering React on the server (we can [use it with hapi](https://github.com/zeit/next.js/tree/master/examples/custom-server-hapi)). And for styling components, Next.js recommends [styled-jsx](https://github.com/zeit/styled-jsx) which we can try - seems like it should work okay with Semantic UI React.
